drivers/serial/uart_uwp.c

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright (c) 2018, UNISOC Incorporated
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <uart.h>
+
+#include "uwp_hal.h"
+#include "uart_uwp.h"
+#include "../interrupt_controller/intc_uwp.h"
```

Comment:
Is this header needed?

Reply:
This header file is needed by uart driver.
There is a 2-Level interrupt mechanism in this chip. UART2 interrupt line connect to interrupt controller, and interrupt controller connect to NVIC. Therefor, interrupt controller need to provide interfaces to register uart isr callback, such as uwp_aon_intc_set_irq_callback.